rpc_jobs/build_summary.yml

```diff
@@ -15,24 +15,9 @@
     dsl: |
       library "rpc-gating@${RPC_GATING_BRANCH}"
       common.globalWraps(){
+        String RE_JOB_REPO_NAME = "rpc-gating"
         dir("rpc-gating/scripts/build_summary"){
-          withCredentials([
-            usernamePassword(
-              credentialsId: "dockerhubrpcjirasvc",
-              usernameVariable: "DOCKERHUB_USER",
-              passwordVariable: "DOCKERHUB_PASSWORD"
-            )
-          ]){
-            sh """#!/bin/bash -xeu
-              # add docker registry creds.
-              # using creds in jenkins global config along with image.push()
-              # doesn't seem to work :(
-              # ok to use single quotes as the \$ is interpreted by groovy not bash
-              docker login --password '${env.DOCKERHUB_PASSWORD}' --username '${env.DOCKERHUB_USER}'
-              docker build . -t "${DOCKER_IMAGE}"
-              docker push "${DOCKER_IMAGE}"
-            """
-          }
+          docker_utils.pullOrBuild(DOCKER_IMAGE, "latest")
```

Comment:
If I understand correctly, this will result in an image being built once, then fetched on all subsequent occasions, even if the code has changed? The image name consists of registry,repo,name & tag. None of these change so what would cause the image to not be available and get rebuilt?

Also you could delete this job, and call pullorbuild in the build-summary job.